README.md

```diff
@@ -1035,10 +1037,88 @@ If your modules have dependencies between them—for example, you can't deploy t
 deployed—you'll need to express those dependencies in your Terragrunt configuration as explained in the next section.
 
 
-#### Dependencies between modules
+#### Passing outputs between modules
 
 Consider the following file structure:
 
+```
+root
+├── backend-app
+│   ├── main.tf
+│   └── terragrunt.hcl
+├── mysql
+│   ├── main.tf
+│   └── terragrunt.hcl
+├── redis
+│   ├── main.tf
+│   └── terragrunt.hcl
+└── vpc
+    ├── main.tf
+    └── terragrunt.hcl
+```
+
+Suppose that you wanted to pass in the VPC ID of the VPC that is created from the `vpc` module in the folder structure
+above to the `mysql` module as an input variable. Or if you wanted to pass in the subnet IDs of the private subnet that
+is allocated as part of the `vpc` module.
+
+You can use the `terragrunt_output` block to extract the output variables to access another modules output variables in
+the terragrunt `inputs` attribute.
+
+For example, suppose the `vpc` module outputs the ID under the name `vpc_id`. To access that output, you would specify
+in `mysql/terragrunt.hcl`:
+
+```
+terragrunt_output "vpc" {
+  config_path = "../vpc"
+}
+
+inputs = {
+  vpc_id = terragrunt_output.vpc.vpc_id
+}
+```
+
+When you apply this module, the output will be read from the `vpc` module and passed in as an input to the `mysql`
+module right before calling `terraform apply`.
+
+You can also specify multiple `terragrunt_output` blocks to access multiple different module output variables. For
+example, in the above folder structure, you might want to reference the `domain` output of the `redis` and `mysql`
+modules for use as `inputs` in the `backend-app` module. To access those outputs, you would specify in
+`backend-app/terragrunt.hcl`:
+
+```
+terragrunt_output "mysql" {
+  config_path = "../mysql"
+}
+
+terragrunt_output "redis" {
+  config_path = "../redis"
+}
```

Comment:
This is very, very cool 👍 

Now that I've seen this, I can't wonder if this should be:

```hcl
dependency "mysql" {
  config_path = "../mysql"
}

dependency "redis" {
  config_path = "../redis"
}

inputs = {
  mysql_url = dependency.mysql.outputs.domain
  redis_url = dependency.redis.outputs.domain
}
```

For this PR, the only difference would be a slight naming change: `terragrunt_output` becomes `dependency` and you read outputs from it via the `.outputs` attribute. However, in a future PR, we could allow you to read _any_ part of that module's config! E.g., Perhaps `dependency.mysql.inputs.foo` would return the value of the "foo" output from the `mysql` dependency and `dependency.redis.remote_state.config.bucket` would return the bucket configured for remote state storage in the `redis` dependency.

Reply:
I like this. I went through and made this change.

Reply:
fca4ac2